tests/phenotype/test_phenotype_resources.py

```diff
@@ -0,0 +1,395 @@
+        kf_id = 'non-existent'
+        body = {}
+        response = self.client.patch(url_for(PHENOTYPES_URL,
+                                           kf_id=kf_id),
+                                   headers=self._api_headers(),
+                                   data=json.dumps(body))
+        # Check status code
+        self.assertEqual(response.status_code, 404)
+        # Check response body
+        response = json.loads(response.data.decode("utf-8"))
+        # Check error message
+        message = 'could not find phenotype'
+        self.assertIn(message, response['_status']['message'])
+        # Check database
+        c = Phenotype.query.filter_by(kf_id=kf_id).count()
+        self.assertEqual(c, 0)
+
+    def test_patch_bad_input(self):
+        """
+        Test update existing phenotype with bad input
+
+        Participant with participant_id does not exist
+        """
+        # Create and save phenotype to db
+        kwargs = self._create_save_to_db()
+
+        # Send patch request
+        body = {
+            'participant_id': 'AAAA1111'
+        }
+        response = self.client.patch(url_for(PHENOTYPES_URL,
+                                           kf_id=kwargs['kf_id']),
+                                   headers=self._api_headers(),
+                                   data=json.dumps(body))
+        # Check status code
+        self.assertEqual(response.status_code, 400)
+        # Check response body
+        response = json.loads(response.data.decode("utf-8"))
+        # Check error message
+        message = 'participant "AAAA1111" does not exist'
+        self.assertIn(message, response['_status']['message'])
+        # Check field values
+        p = Phenotype.query.first()
+        self.assertEqual(p.phenotype, kwargs.get('phenotype'))
+        self.assertEqual(p.age_at_event_days, kwargs.get('age_at_event_days'))
+
+    def test_patch_missing_req_params(self):
+        """
+        Test create phenotype that is missing required parameters in body
+        """
+        # Create and save phenotype to db
+        kwargs = self._create_save_to_db()
+        # Create phenotype data
+        body = {
+            'phenotype': 'Hand Tremor'
+        }
+        # Send patch request
+        response = self.client.patch(url_for(PHENOTYPES_URL,
+                                           kf_id=kwargs['kf_id']),
+                                   headers=self._api_headers(),
+                                   data=json.dumps(body))
+        # Check status code
+        self.assertEqual(response.status_code, 400)
+        # Check response body
+        response = json.loads(response.data.decode("utf-8"))
+        # Check error message
+        message = 'could not update phenotype'
+        self.assertIn(message, response['_status']['message'])
+        # Check field values
+        p = Phenotype.query.first()
+        self.assertEqual(p.phenotype, kwargs['phenotype'])
+
+    def test_delete(self):
```

Comment:
This is basically the same test as the one below without the checking of response code? We could probably get rid of it.